README.md

```diff
@@ -1,307 +1,212 @@
 
-```python
-import psycopg2
+---
 
-from bytewax import operators as op
-from bytewax.outputs import FixedPartitionedSink, StatefulSinkPartition
+## Local Development, Testing, and Production
 
+**Local Development:**
+- Use `TestingSource` and `inspect` operators for debugging.
+- Iterate quickly by running your flow with `python -m bytewax.run my_flow.py`.
+- Develop custom connectors and sinks locally with Python tooling you already know.
 
-class PsqlSink(StatefulSinkPartition):
-    def __init__(self):
-        self.conn = psycopg2.connect("dbname=website user=bytewax")
-        self.conn.set_session(autocommit=True)
-        self.cur = self.conn.cursor()
+**Testing:**
+- Integration tests: Use `TestingSource` and run flows directly in CI environments.
+- Unit tests: Test individual functions and operators as normal Python code.
+- [More on Testing](https://docs.bytewax.io/stable/guide/contributing/testing.html)
 
-    def write_batch(self, values):
-        query_string = """
-            INSERT INTO events (user_id, data)
-            VALUES (%s, %s)
-            ON CONFLICT (user_id)
-            DO UPDATE SET data = %s;
-        """
-        self.cur.execute_values(query_string, values)
+**Production:**
+- Scale horizontally by running multiple workers on multiple machines.
+- Integrate with Kubernetes for dynamic scaling, monitoring, and resilience.
+- Utilize `waxctl` for standardized deployments and lifecycle management.
 
-    def snapshot(self):
-        pass
+---
 
-    def close(self):
-        self.conn.close()
+## Deployment Options
 
+### Running Locally
 
-class PsqlOutput(FixedPartitionedSink):
-    def list_parts(self):
-        return ["single"]
+For experimentation and small-scale jobs:
 
-    def build_part(step_id, for_part, resume_state):
-        return PsqlSink()
+```sh
+python -m bytewax.run my_dataflow.py
 ```
 
-#### Execution
-
-Bytewax dataflows can be executed in a single Python process, or on multiple processes on multiple hosts with multiple worker threads. When processing data in a distributed fashion, Bytewax uses routing keys to ensure your state is updated in a correct way automatically.
+Multiple workers and threads:
 
 ```sh
-# a single worker locally
-python -m bytewax.run my_dataflow:flow
-
-# Start two worker threads in a single process.
-python -m bytewax.run my_dataflow -w 2
-
-# Start a process on two separate machines to form a Bytewax cluster.
-# Start the first process with two worker threads on `machine_one`.
-machine_one$ python -m bytewax.run my_dataflow -w 2 -i0 -a "machine_one:2101;machine_two:2101"
-
-# Then start the second process with three worker threads on `machine_two`.
-machine_two$ python -m bytewax.run my_dataflow -w 3 -i1 -a "machine_one:2101;machine_two:2101"
+python -m bytewax.run my_dataflow.py -w 2
 ```
 
```

Comment:
Not sure we want to have this in the readme, but there's also the testing runner to easily try multiple processes locally:
```
python -m bytewax.testing my_dataflow -w1 -p2
```

It could be worth mentioning because often you add multiple workers with `-w2` and not see improvements if all the workers are contending the GIL.